api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+        }
+
+        try {
+            jacksonFactory = JacksonFactory.getDefaultInstance();
+            httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (Exception  e) {
+            e.printStackTrace();
+            throw new InitializationException("Trouble at initializing", e);
+        }
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        return Arrays.asList(new ApplicationReadyEvent(), new DocumentUpdatedEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        log.info("Event triggered: " + event + " with source " + source);
+        boolean applicationStarted = false, configChanged = false;
+        if (event instanceof ApplicationReadyEvent) {
+            applicationStarted = true;
+        }
+        if (event instanceof DocumentUpdatedEvent) {
+            XWikiDocument document = (XWikiDocument) source;
+            configDocRef = getConfigDocRef();
+            if (document != null && document.getDocumentReference().compareTo(configDocRef) == 0) {
+                configChanged = true;
+            }
+        }
+
+        if(configChanged) {
+            log.info("Reloading config.");
+            readConfigDoc(xwikiContextProvider.get());
+        }
+
+        if(applicationStarted || configChanged) {
+            try {
+                initialize();
+            } catch (InitializationException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    // internals
+
+    private GoogleAppsAuthServiceImpl authService;
+
+    private DocumentReference configDocRef;
+    private ObjectReference configObjRef;
+
+    /** A map of hash to full redirects. */
+    private Map<String, String> storedStates = new HashMap<>();
+
+    private FileDataStoreFactory dsFactory;
+
+    private JacksonFactory jacksonFactory;
+
+    private NetHttpTransport httpTransport;
+
+    private CloseableHttpClient httpclient = HttpClients.createDefault();
+
+
+    private BaseObject getConfigDoc(XWikiContext context) throws XWikiException
+    {
+        configDocRef = getConfigDocRef();
+        XWikiDocument doc = context.getWiki().getDocument(configObjRef, context);
+        BaseObject result = doc.getXObject(configObjRef, false, context);
+        if (result  ==  null) {
+            log.warn("Can't access Config document.");
+        }
+        return result;
+    }
+
+    private Boolean configActiveFlag;
+    private Boolean useCookies;
+    private Boolean skipLoginPage;
+    private Boolean authWithCookies;
+    private String configAppName;
+    private String configClientId;
+    private String configClientSecret;
+    private String configDomain;
+
+    private Boolean configScopeUseAvatar;
+    private Boolean configScopeUseDrive;
+    private Long configCookiesTTL;
+
+    // ---------------------------- constants ---------------------------------------------
```

Comment:
At this point, I read a lot of code in this class and makes me think that there is too much and this class should be split.

Reply:
That class is definitely planned to be refactored.
No checkstyle will pass without it.